cypress/e2e/spec.cy.ts

```diff
@@ -0,0 +1,68 @@
+describe("E2E Test", () => {
+  it("should visit Discover Page Change Genre", () => {
+    cy.visit("/");
+    cy.url().should("include", "/");
+
+    cy.get("button").click();
+    cy.get("span").contains("Dance").click();
+    cy.get("button").should("contain", "Dance");
```

Comment:
See if you can be more precise when querying elements... This looks like a brittle test that can start failing if someone decides to add another button or span element on that page. I can see that for some tests you are using `[data-testid=` and that approach is much better as it won't start failing if someone adds something unrelated to the page or moves stuff around. At least try to specify a parent element that will hold your targeted elements to be more precise in selecting an element.